File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        if(s.length() < 2) return false;
        stack<int> pStack;
        for(int i=0; i<s.length(); i++){
            if(s[i] == '(' || s[i] == '{' || s[i] == '[') pStack.push(s[i]);
            else if(s[i] == ')'){
                if(pStack.empty()) return false;
                else if (pStack.top() != '(') return false;
                else pStack.pop();
            }
            else if(s[i] == '}'){
                if(pStack.empty()) return false;
                else if (pStack.top() != '{') return false;
                else pStack.pop();
            }
            else if(s[i] == ']'){
                if(pStack.empty()) return false;
                else if (pStack.top() != '[') return false;
                else pStack.pop();
            }
            else continue;
        }

        if(!pStack.empty()) return false;
        return true;
    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    if(s.length%2 !== 0) return false;
    let stack = [];
    let map = {
        '(': ')',
        '{': '}',
        '[': ']'
    }
    for(let i = 0; i <s.length; i++){
        if(s[i] === '(' || s[i] === '{' || s[i] === '['){
            stack.push(s[i]);
        } else if(s[i] !== map[stack.pop()]){
                return false;
        }
    }
    return !stack.length;
};